docs/src/proposals/off-chain-message-signing.md

```diff
@@ -0,0 +1,78 @@
+# Off-chain message signing
+## Motivation
+
+There is ecosystem demand for a method of signing non-transaction messages with
+a Solana wallet. Typically this is some kind of "proof of wallet ownership" for
+entry into a whitelisted system.
+
+Some inspiration can be gleaned from relevant portions of Ethereum's
+[EIP-712](https://eips.ethereum.org/EIPS/eip-712)
+
+## Considerations
+
+* Security
+  * Off-chain message signatures should not be valid transaction message signatures
+* Future-proofing
+  * Versioning
+  * Co-exist with versioned transaction messages and extended length transactions
+* Compatibility
+  * Hardware wallet signing
+  * Localization
+
+## Payload format
+
+| 0x00  ..= 0x0f | 0x10           | 0x11           | 0x12  ..= 0x13 | 0x14 .. |
+| :------------: | :------------: | :------------: | :------------: | :-----: |
+| Signing Domain | Header Version | Message Format | Message Length | Message |
+
+The payload consists of a 16-byte signing domain specifier, 4-byte header and
+n-byte message.
+
+### Signing Domain Specifier
+The signing domain specifier is a prefix byte string used to give similar structure
+to all off-chain message signatures. We assign its value to be:
+```
+b"\xffsolana offchain"
+```
+The first byte, `\xff`, was chosen for the following reasons
+1. It corresponds to a value that is illegal as the first byte in a transaction
```

Comment:
Versioned transaction messages always set the first bit and then use the remaining bits of the first byte to determine a version number for the message. This `\xff` value corresponds to version 128 and so we should ensure that the version encoding for transactions spills over into another byte if higher versions are ever used.

Reply:
Never mind I see that you covered this in the "Runtime Considerations" section below